Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -409,8 +408,8 @@ void VKTexture::Load(u32 level, u32 width, u32 height, u32 row_length, const u8*
   else
   {
     // Create a temporary staging buffer that is destroyed after the image is copied.
-    temp_buffer = StagingBuffer::Create(STAGING_BUFFER_TYPE_UPLOAD, upload_size,
-                                        VK_BUFFER_USAGE_TRANSFER_SRC_BIT);
+    std::unique_ptr<StagingBuffer> temp_buffer = StagingBuffer::Create(
+        STAGING_BUFFER_TYPE_UPLOAD, upload_size, VK_BUFFER_USAGE_TRANSFER_SRC_BIT);
```

Comment:
This affects object lifetime, not sure whether that matters but just noting...

Reply:
In fact, I'm pretty sure this breaks the code, looking at this in more detail. `upload_buffer` keeps a pointer to the now out-of-scope `temp_buffer->GetBuffer()`.

Reply:
As @AdmiralCurtiss said, `upload_buffer` will point to a destroyed object _(held by the unique_ptr)_ when exiting the current `else` statement _(i.e. which will destroy `temp_buffer` declared in this scope)_.

Reply:
Since it wasn't caught when building I didn't think it would be an issue.

Do you have any tips for how to avoid this in the future? (Other than switching to Rust :P)